### ISSUE TITLE
distutils packager/installer includes admin tools not meant to be included

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5042]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5042|
|Type|defect|
|Created|2011-04-08 22:45:20Z|

`bin/admin/` includes release management tools.  These are not intended to be released.  However, they're found by setup.py and included in the release.

Attachments:

* [exclude-admin-5042.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/474/474ad11cb138a598fc50c608977091868bec40c1/6b868efb6aa2a8d9631272bfd701617c9c82cc69.patch) (861 bytes) - added by thijs on 2011-05-13 16:08:03Z - only include listed twisted projects

<details><summary>Searchable metadata</summary>

```
trac-id__5042 5042
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__Twisted_11_1 Twisted-11.1
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__release_management release management
keywords__ 
time__1302302720000000 1302302720000000
changetime__1307311997000000 1307311997000000
version__None None
owner__thijs thijs
cc__radix cc__thijs cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

`admin` is identified as a subproject by the top-level `setup.py`, instead it should be skipped I suppose.  There's a list of valid subprojects in `twisted.python.dist`.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

Attached patch only includes sub-projects in `bin` that are defined in `twisted.python.dist.twisted_subprojects`.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_7"></a>|@jesstess set owner to @thijstriemstra|
|-|-|

Thanks for working on this, thijs. I didn't cut a release to do an end-to-end test of this, but the changes look good, so if you're happy with how much it's been tested please merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [32049]) Apply exclude-admin-5042.patch: setup.py now only includes registered
Twisted sub-projects, and ignores other directories, like bin/admin,
that contain various tools that should not be included in a release.

Author: thijs
Reviewer: jesstess
Fixes: [#5042](https://github.com/twisted/twisted/issues/5042)
